src/actions/context.ts

```diff
@@ -1,170 +1,64 @@
-	const loadingSpinner = ora('Resolving package versions...').start();
-	try {
-		const [stylelintVersion, stylelintConfigVersion] = await Promise.all([
-			resolvePackageVersion('stylelint'),
-			resolvePackageVersion('stylelint-config-standard'),
-		]);
-		loadingSpinner.succeed('Successfully resolved package versions');
-		return { stylelint: stylelintVersion, stylelintConfig: stylelintConfigVersion };
-	} catch (error) {
-		loadingSpinner.fail('Failed to resolve package versions');
-		throw error;
-	}
-}
-
-async function handleSpecialFlags(flags: Flags, exit: (code: number) => never): Promise<void> {
-	if (flags['--help'] || flags['-h']) {
-		showHelp();
-		exit(0);
-	}
-
-	if (flags['--version'] || flags['-v']) {
-		try {
-			const version = await resolvePackageVersion('create-stylelint');
-			log(`create-stylelint v${version}`);
-			exit(0);
-		} catch (error) {
-			log(red('Failed to fetch package version'));
-			log(dim('Please check your network connection and try again.\n'));
-			exit(1);
-		}
-	}
-}
-
-export async function createContext(
-	args: string[],
-	exit: (code: number) => never = process.exit,
-): Promise<Context> {
-	const flags = parseFlags(args);
-	validateFlags(flags, exit);
-	await handleSpecialFlags(flags, exit);
-
-	let packageManager: PackageManager | null = null;
-	if (flags['--use-npm']) packageManager = 'npm';
-	if (flags['--use-pnpm']) packageManager = 'pnpm';
-	if (flags['--use-yarn']) packageManager = 'yarn';
-	if (flags['--use-bun']) packageManager = 'bun';
-
-	if (!flags['--dry-run']) {
-		await validatePackageJson(packageManager);
-	}
-
-	const currentWorkingDirectory = new URL(`${pathToFileURL(process.cwd())}/`);
-
-	let packageVersions = { stylelint: '', stylelintConfig: '' };
-
-	if (!flags['--dry-run']) {
-		try {
-			packageVersions = await resolvePackageVersions();
-		} catch (error) {
-			log(red(`[ERROR] ${error}`));
-			log(red('[ERROR] Failed to resolve package versions'));
-			newline();
-			exit(1);
-		}
-	}
+	const packageManagerFlags = {
+		'--use-npm': flags['--use-npm'],
+		'--use-yarn': flags['--use-yarn'],
+		'--use-pnpm': flags['--use-pnpm'],
+		'--use-bun': flags['--use-bun'],
+		'--use-deno': flags['--use-deno'],
+	};
 
 	return {
-		currentWorkingDirectory,
-		exit,
-		help: flags['--help'] ?? false,
-		isDryRun: flags['--dry-run'],
-		packageManager: packageManager || 'npm',
-		packageVersions,
+		help: flags['--help'] || false,
+		version: flags['--version'] || false,
 		prompt: prompts,
-		shouldSkipInstall: flags['--skip-install'] ?? false,
-		useBun: flags['--use-bun'],
-		useNpm: flags['--use-npm'],
-		usePnpm: flags['--use-pnpm'],
-		useYarn: flags['--use-yarn'],
+		isDryRun: Boolean(flags['--dry-run']),
+		shouldSkipInstall: Boolean(flags['--no-install']),
+		cwd: new URL(`${pathToFileURL(process.cwd())}/`),
+		packageManager: await getPackageManager(packageManagerFlags),
+		exit: (code) => process.exit(code),
```

Comment:
question

I find it kinda weird that you are not using `process.exit` directly.
I was expecting that the argument would always be the same or something that would justify putting it on `context`.
Why did you do it?

Reply:
Well I had to drill several functions to find whether that `exit` was exiting the program or not and what it was doing differently compared to `process.exit`.

It might be useful someday if we choose to use `exitCode` instead.